src/shogun-core-main/src/main/java/de/terrestris/shoguncore/security/access/entity/PersistentObjectPermissionEvaluator.java

```diff
@@ -98,22 +98,16 @@ protected PermissionCollection extractUserPermissions(User user,
     /**
      * @param groupPermissionsMap
      */
-    protected PermissionCollection extractGroupPermissions(User user,
-                                                           Map<UserGroup, PermissionCollection> groupPermissionsMap) {
-
-        Set<Permission> aggregatedGroupPermissions = new HashSet<Permission>();
-
-        Set<UserGroup> userGroupsWithPermissions = groupPermissionsMap.keySet();
-
-        for (UserGroup userGroup : userGroupsWithPermissions) {
+    protected PermissionCollection extractGroupPermissions(User user, Map<UserGroup, PermissionCollection> groupPermissionsMap) {
+        Set<Permission> aggregatedGroupPermissions = new HashSet<>();
 
+        for (Map.Entry<UserGroup, PermissionCollection> userGroupEntry: groupPermissionsMap.entrySet()) {
+            UserGroup userGroup = userGroupEntry.getKey();
             if (userGroup.getMembers().contains(user)) {
-                Set<Permission> groupPermissions = groupPermissionsMap.get(userGroup).getPermissions();
+                Set<Permission> groupPermissions =userGroupEntry.getValue().getPermissions();
```

Comment:
```suggestion
                Set<Permission> groupPermissions = userGroupEntry.getValue().getPermissions();
```

Reply:
:+1: I've added further checkstyle rule`<module name="WhitespaceAround"/>` to check for this as well.